src/modules/logicModule/paymentClient/ERC20PaymentClient.sol

```diff
@@ -189,20 +164,64 @@ abstract contract PaymentClient is IPaymentClient, ContextUpgradeable {
 
     function _ensureValidRecipient(address recipient) private view {
         if (recipient == address(0) || recipient == address(this)) {
-            revert Module__PaymentClient__InvalidRecipient();
+            revert Module__ERC20PaymentClient__InvalidRecipient();
         }
     }
 
     function _ensureValidAmount(uint amount) private pure {
-        if (amount == 0) revert Module__PaymentClient__InvalidAmount();
+        if (amount == 0) revert Module__ERC20PaymentClient__InvalidAmount();
     }
 
     function _ensureValidPaymentOrder(PaymentOrder memory order) private view {
         if (order.amount == 0) {
-            revert Module__PaymentClient__InvalidAmount();
+            revert Module__ERC20PaymentClient__InvalidAmount();
         }
         if (order.recipient == address(0) || order.recipient == address(this)) {
-            revert Module__PaymentClient__InvalidRecipient();
+            revert Module__ERC20PaymentClient__InvalidRecipient();
         }
     }
+
+    //--------------------------------------------------------------------------
+    // {ERC20PaymentClient} Function Implementations
+
+    /// @dev Ensures `amount` of payment tokens exist in address(this).
+    function _ensureTokenBalance(uint amount) internal virtual {
+        // Trigger callback from orchestrator to transfer tokens
```

Comment:
Yeah your right.
The initial purpose of the ```amount - balance``` was to prevent that.
The problem here is with the creation of a "vent" to drain funds we come into the risk of not guaranteeing that payments can go out.
In case of -> Are never claimed by the rightful owner:
This is in my opinion the correct behavior. Even if the address doesnt claim the funds it still has a right to do so.
In case of -> tokens that are sent directly to the contract:
Totally agree.

In general the problem with these tokens that lay around and do nothing is, that they are perceived as "useless".
Maybe a mechanism to put them to use (staking or other) may be a way to go here.
I will talk to the others about this.